Initial batch script

workentin opened the pull request:
Only report formatting and some system info checks are ported from PSv3 script so far.

workentin committed:
Initial batch script

Only report formatting and some system info checks are ported from PSv3 script so far.

workentin committed:
Update chaps.bat

workentin committed:
Update chaps.bat sysinfo checks and add debugging control

workentin committed:
Update chaps.bat with BitLocker, AutoUpdate and SMB  checks

workentin committed:
Updated CHAPS.bat IP check for Windows XP

Made check more robust for checking networking information.

workentin committed:
Updated SMB checks

Verified working on Windows XP and Windows 7 SP1.

workentin committed:
Update chaps.bat

Added local administrators group check and AlwaysInstallElevated checks.

workentin committed:
Updated READABLE_DATE for footer section

workentin committed:
Moved CutSec Footer to function call

cutaway commented:
Setting TESTING to false does not turn everything off.

```cmd
C:\Users\student\Downloads>.\chaps.bat
##########################
# Saving output to: C:\Users\student\AppData\Local\Temp\SANS-CORE-VM_20251805_195329\SANS-CORE-VM_chaps.txt
##########################
[*] Collecting systeminfo to: C:\Users\student\AppData\Local\Temp\SANS-CORE-VM_20251805_195329\SANS-CORE-VM_sysinfo.txt
[DEBUG] Setting IPv4_ADDR to: " 192.168.85.140"
[DEBUG] Set IPv4_ADDR: 192.168.85.140
[DEBUG] Finished checking for "IPv4 Address" - IPv4_ADDR: 192.168.85.140
# Report saved to: C:\Users\student\AppData\Local\Temp\SANS-CORE-VM_20251805_195329\SANS-CORE-VM_chaps.txt
```

cutaway commented:
WMIC outputs appear to remove the starting ```[``` from some outputs. I could not find the issue.

```cmd
[DEBUG] Called GetWMICValue
[DEBUG] WMIC_CLASS: os
[DEBUG] WMIC_PROPERTY: OSArchitecture
[DEBUG] WMIC_EXE: C:\Windows\system32\wbem\wmic.exe
[DEBUG] Attempting: wmic os get OSArchitecture
[DEBUG] WMIC Result: 64-bit
"DEBUG] Returning RESULT: "64-bit
[DEBUG] Setting OS_ARCH: 64-bit
[DEBUG] Attempting to get Domain/Workgroup
[DEBUG] Calling GetWMICValue computersystem Domain
[DEBUG] Called GetWMICValue
[DEBUG] WMIC_CLASS: computersystem
[DEBUG] WMIC_PROPERTY: Domain
[DEBUG] WMIC_EXE: C:\Windows\system32\wbem\wmic.exe
[DEBUG] Attempting: wmic computersystem get Domain
[DEBUG] WMIC Result: WORKGROUP
"DEBUG] Returning RESULT: "WORKGROUP
[DEBUG] Setting DOMAIN: WORKGROUP
[*] Windows Product: Microsoft Windows 10 Enterprise
[*] OS Version: 10.0.19045
[*] OS Architecture: 64-bit
[*] System Type: x64-based PC
[*] Domain/Workgroup: WORKGROUP
```

cutaway commented:
I want to move the outputs to the local directory instead of ```AppData```. It'll just get lost there.

```cmd
# Report saved to: C:\Users\student\AppData\Local\Temp\SANS-CORE-VM_20251805_195329\SANS-CORE-VM_chaps.txt
```

workentin committed:
Add CheckRestrictAnonymous

workentin committed:
Added NTLM Sec, RPC Remote Login, and Cached Logons Checks

workentin committed:
Made checks modularized

Added flags for enable/disable checks
Put WMIC queries into function call
Put Admin Rights check to function call
Put check time into funciton call

workentin committed:
Moved Print Footer to function call

cutaway commented:
Thanks, @workentin. This PR's content from `cmd-bat-refactor` was merged into the `claude-dev` development branch as part of the v2.0 consolidation, and then significantly extended to cover all 63 canonical checks. The current `CMD/chaps.bat` on `claude-dev` is based on your work.

Closing this PR since the branch is being retired as part of cleaning up stale branches before the v2.0 release. The commit history is preserved in the `claude-dev` merge and will flow to `main` at release.
